src/System.Text.Json/tests/Json.cs

```diff
@@ -9,6 +9,132 @@ public class Json
 
         public Array Array { get; set; }
 
+        public Value Value { get; set; }
+
+        public Json this[string index]
+        {
+            get
+            {
+                var json = new Json();
+                if (Object == null) throw new NullReferenceException();
+                if (Object.Pairs == null) throw new NullReferenceException();
+
+                foreach (var pair in Object.Pairs)
+                {
+                    if (pair.Name == index)
+                    {
+                        switch (pair.Value.Type)
+                        {
+                            case Value.ValueType.Object:
+                                json.Object = pair.Value.ObjectValue;
+                                break;
+                            case Value.ValueType.Array:
+                                json.Array = pair.Value.ArrayValue;
+                                break;
+                            case Value.ValueType.String:
+                            case Value.ValueType.Number:
+                            case Value.ValueType.False:
+                            case Value.ValueType.True:
+                            case Value.ValueType.Null:
+                                json.Value = pair.Value;
+                                break;
+                            default:
+                                break;
+                        }
+
+                        return json;
+                    }
+                }
+
+                throw new KeyNotFoundException();
+            }
+        }
+
+        public Json this[int index]
+        {
+            get
+            {
+                var json = new Json();
+                if (Array == null) throw new NullReferenceException();
+                if (Array.Values == null) throw new NullReferenceException();
+                var values = Array.Values;
+                if (index < 0 || index >= values.Count) throw new IndexOutOfRangeException();
+                var value = values[index];
```

Comment:
I would state List<Value> since It seems like it would be Array<Value> not List<Value>